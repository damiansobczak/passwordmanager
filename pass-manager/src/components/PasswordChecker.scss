.password-strength-meter {
  overflow: hidden;
  height: 0;
  transition: all 0.2s ease-in-out;

  &.active {
    height: 20px;
    transition: all 0.2s ease-in-out;
  }
}

.password-strength-meter-progress {
  width: 100%;
  -webkit-appearance: none;
  appearance: none;
  height: 8px;
  border-radius: 5px;

  &::-webkit-progress-bar {
    background-color: #f3f3f3;
  }

  &::-webkit-progress-value {
    border-radius: 10px;
  }
}

.strength {
  &-Weak::-webkit-progress-value {
    background-color: #d32f2f;
  }
  &-Fair::-webkit-progress-value {
    background-color: #ef6c00;
  }
  &-Good::-webkit-progress-value {
    background-color: #fbc02d;
  }
  &-Strong::-webkit-progress-value {
    background-color: #4caf50;
  }
}
