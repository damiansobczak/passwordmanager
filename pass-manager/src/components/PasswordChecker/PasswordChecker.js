import React, { Component } from "react";
import zxcvbn from "zxcvbn";
import "./PasswordChecker.scss";
import PropTypes from "prop-types";

class PasswordChecker extends Component {
  createPasswordLabel = result => {
    switch (result.score) {
      case 0:
        return "Weak";
      case 1:
        return "Weak";
      case 2:
        return "Fair";
      case 3:
        return "Good";
      case 4:
        return "Strong";
      default:
        return "Weak";
    }
  };

  render() {
    const password = this.props.password;
    const testedPassword = zxcvbn(password);
    return (
      <div className={`form card-form-inputs password-strength-meter ${password ? "active" : ""}`}>
        <div className="input-field col s12">
          <div className="password-strength-meter-wrapper">
            <progress className={`password-strength-meter-progress strength-${this.createPasswordLabel(testedPassword)}`} value={testedPassword.score} max="4" />
          </div>
        </div>
      </div>
    );
  }
}

PasswordChecker.propTypes = {
  password: PropTypes.string
};

export default PasswordChecker;
