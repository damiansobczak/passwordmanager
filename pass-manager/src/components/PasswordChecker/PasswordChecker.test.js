import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PasswordChecker from "./PasswordChecker";

describe("<PasswordChecker />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PasswordChecker password="$hashpassWd1234" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check password strength", () => {
    const wrapper = shallow(<PasswordChecker password="$hashpassWd1234" />);
    expect(
      wrapper
        .find(".password-strength-meter-progress")
        .props()
        .className.includes("strength-Strong")
    ).toBe(true);
  });
});
