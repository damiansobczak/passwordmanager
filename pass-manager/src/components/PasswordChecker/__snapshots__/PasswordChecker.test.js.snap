// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordChecker /> renders correctly 1`] = `
<div
  className="form card-form-inputs password-strength-meter active"
>
  <div
    className="input-field col s12"
  >
    <div
      className="password-strength-meter-wrapper"
    >
      <progress
        className="password-strength-meter-progress strength-Strong"
        max="4"
        value={4}
      />
    </div>
  </div>
</div>
`;
